crates/bevy_ecs/src/world/mod.rs

```diff
@@ -190,6 +190,13 @@ impl World {
         Ok(component_id)
     }
 
+    pub(crate) fn register_or_get_id<T: Component>(&mut self) -> ComponentId {
+        self.register_component(ComponentDescriptor::new::<T>())
```

Comment:
I dont think we should be constructing a ComponentDescriptor (which allocates a String) every time something needs a ComponentId.

Ideally we only initialize the descriptor if the component hasn't already been registered.